File: ios/Theme/ResultScreen.swift
public class ResultScreen: ViewStyle {
  private static let KEY: String = "resultScreen"
  private let style: Style
  private let target: NSDictionary?
  private let font: Font
  private let color: Color
  private let resultAnimation: ResultAnimation
  private let sessionAbortAnimation: SessionAbortAnimation

  init() {
    self.style = Style()
    self.font = Font()
    self.color = Color()

    self.target = self.style.getTarget(ResultScreen.KEY)

    self.resultAnimation = ResultAnimation(target: self.target)
    self.sessionAbortAnimation = SessionAbortAnimation(target: self.target)

    super.init(key: ResultScreen.KEY)
  }

  override public func getForegroundColor() -> UIColor {
    return self.getForegroundColor(defaultColor: "#272937")
  }

  public func getActivityIndicatorColor() -> UIColor {
    return self.color.getColor(self.target, key: "activityIndicatorColor", defaultColor: "#026ff4")
  }

  public func getUploadProgressFillColor() -> UIColor {
    return self.color.getColor(self.target, key: "uploadProgressFillColor", defaultColor: "#026ff4")
  }

  public func getUploadProgressTrackColor() -> UIColor {
    return self.color.getColor(
      self.target, key: "uploadProgressTrackColor", defaultColor: "#b3d4fc")
  }

  public func getShowUploadProgressBar() -> Bool {
    let key = "isShowUploadProgressBar"

    if !self.style.exists(self.target, key: key) {
      return true
    }

    return self.target?[key] as! Bool
  }

  public func getFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "font")
  }

  public func getResultAnimation() -> ResultAnimation {
    return self.resultAnimation
  }

  public func getSessionAbortAnimation() -> SessionAbortAnimation {
    return self.sessionAbortAnimation
  }
}

File: android/src/main/java/com/azifacemobile/theme/ResultScreen.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

import javax.annotation.Nullable;

public class ResultScreen extends ViewStyle {
  private static final String KEY = "resultScreen";
  private final Theme theme;
  private final JSONObject target;
  private final Font font;
  private final Color color;
  private final ResultAnimation resultAnimation;
  private final SessionAbortAnimation sessionAbortAnimation;

  public ResultScreen(ReactApplicationContext context) {
    super(KEY);

    this.theme = new Theme();
    this.font = new Font(context);
    this.color = new Color();

    this.target = this.theme.getTarget(KEY);

    this.resultAnimation = new ResultAnimation(context, this.target);
    this.sessionAbortAnimation = new SessionAbortAnimation(context, this.target);
  }

  @Override
  public int getForegroundColor() {
    return super.getForegroundColor("#272937");
  }

  public int getActivityIndicatorColor() {
    return this.color.getColor(this.target, "activityIndicatorColor", "#026ff4");
  }

  public int getUploadProgressFillColor() {
    return this.color.getColor(this.target, "uploadProgressFillColor", "#026ff4");
  }

  public int getUploadProgressTrackColor() {
    return this.color.getColor(this.target, "uploadProgressTrackColor", "#b3d4fc");
  }

  public boolean getShowUploadProgressBar() {
    final String key = "isShowUploadProgressBar";
    try {
      if (!this.theme.exists(this.target, key)) {
        return true;
      }

      return this.target.getBoolean(key);
    } catch (JSONException e) {
      return true;
    }
  }

  @Nullable
  public Typeface getFont() {
    return this.font.getTypography(this.target, "font");
  }

  public ResultAnimation getResultAnimation() {
    return this.resultAnimation;
  }

  public SessionAbortAnimation getSessionAbortAnimation() {
    return this.sessionAbortAnimation;
  }
}
